### PR TITLE
Introducing `staticcheck` linter to detect duplicate package imports in Go code.

### LINKED ISSUE #27676: Add a linter for packages is being imported more than once (ST1019)
There is a linter that can detect packages being imported more than once. Add it to the standard linters in `.github` that run on every PR.

### PR DESCRIPTION
**Pull request by Archisman-Mridha**
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [ ] Thanks for contributing!

> Staticcheck linter has been introduced to the codebase to detect duplicate package imports in Go code.

- [x] A staticcheck.conf file has been created at the repository root. This file tells staticcheck to only consider `ST1019` check (duplicate package imports in Go code).
- [x] Existing duplicate package imports in Go code (detected by staticcheck) have been removed.
- [x] Staticcheck linting step has been added to the `.github/workflows/lint-go.yaml` workflow. Now, whenever a new pull request is opened, staticcheck linter will detect duplicate package imports in Go code and throw errors.

Fixes: #27676

### CODE REVIEW

File: .github/workflows/lint-go.yaml [hunk 1]
```diff
@@ -57,6 +57,24 @@ jobs:
...
+      - name: Install Go
+        uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # v4.1.0
+        with:
+          # renovate: datasource=golang-version depName=go
+          go-version: 1.21.0
+      - name: Checkout code
+        uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
+        with:
+          persist-credentials: false
+      - name: Run staticcheck linting
```
Comment (chancez) [inline .github/workflows/lint-go.yaml:73]: Static lint can be run as part of golangci-lint, do we need to install it separately or can we just enable that lint in our golangci-lint.yaml?

**Review by nebril — CHANGES_REQUESTED** [review]
Thanks for the PR! Code changes look good. Can you go with Chance's suggestion to add staticcheck to ` golangci-lint.yaml` instead of running it separately?

File: .github/workflows/lint-go.yaml (hunk 1 above)
Comment (Archisman-Mridha) [reply .github/workflows/lint-go.yaml:73]: Hey there. I am working on this.

I removed the `staticcheck.conf` file and in the `.golangci.yaml` file, I made this change -
```yaml
staticcheck:
    go: "1.21.0"
    checks: ["ST1019"] # Check for duplicate package imports  
```

To test, whether it works as intended or not, I intentionally created a duplicate package import in the `pkg/ebpf/map_register.go` file. Then I ran `golangci-lint run  --config=./.golangci.yaml ./pkg/ebpf`. But surprisingly no errors were thrown.

**Comment by Archisman-Mridha** [comment]
> Thanks for the PR! Code changes look good. Can you go with Chance's suggestion to add staticcheck to ` golangci-lint.yaml` instead of running it separately?

Working on this (Sorry for the delay).
Also I believe you meant `.golangci.yaml` instead of `golangci-lint.yaml`

File: .github/workflows/lint-go.yaml (hunk 1 above)
Comment (chancez) [reply .github/workflows/lint-go.yaml:73]: Are you able to push your changes to the PR? Hard to tell if it's just a local issue or not.

Comment (lmb) [reply .github/workflows/lint-go.yaml:73]: NB: golangci-lint staticcheck is not the same as the binary. They just copied some rules: https://golangci-lint.run/usage/linters/#staticcheck

Comment (chancez) [reply .github/workflows/lint-go.yaml:73]: Ahhh. Why would they reuse the name like that 🤦 . Okay well then I guess this thread can be resolved.

**Review by chancez — APPROVED** [review]
Okay, sounds like staticcheck in golangci-lint is not the same as standalone staticcheck. The actual mechanical changes LGTM.

**Review by thorn3r — APPROVED** [review]
Nice addition, thanks!

**Review by aanm — APPROVED** [review]

File: staticcheck.conf [hunk 2]
```diff
@@ -0,0 +1 @@
+checks= ["ST1019"]
```
Comment (aanm) [inline staticcheck.conf:1]: Can we move this file to some other place? Perhaps under `.github/`?

**Review by aanm — CHANGES_REQUESTED** [review]

**Review by lmb — APPROVED** [review]

**Review by nebril — APPROVED** [review]

**Review by rastislavs — APPROVED** [review]

**Review by pippolo84 — APPROVED** [review]
Thanks! :rocket:

**Review by joamaki — APPROVED** [review]

File: .github/workflows/lint-go.yaml [hunk 3]
```diff
@@ -57,6 +57,24 @@ jobs:
...
+        run: |
+          go install honnef.co/go/tools/cmd/staticcheck@latest
```
Comment (squeed) [inline .github/workflows/lint-go.yaml:75]: @aanm do we prefer to do `@latest` here, or let renovate do the updating?

Comment (aanm) [reply .github/workflows/lint-go.yaml:75]: Good call, this should be `go install honnef.co/go/tools/cmd/staticcheck@2023.1.6`

**Review by joestringer — CHANGES_REQUESTED** [review]
Thanks for the contribution! @Archisman-Mridha please pin the staticcheck version, add staticcheck.yaml to CODEOWNERS (you can assign it to @cilium/ci-structure) and rebase. 

@cilium/ipsec PTAL

File: pkg/hubble/parser/threefour/parser.go [hunk 4]
```diff
@@ -450,7 +449,7 @@ func decodeSecurityIdentities(dn *monitor.DropNotify, tn *monitor.TraceNotify, p
 func decodeTrafficDirection(srcEP uint32, dn *monitor.DropNotify, tn *monitor.TraceNotify, pvn *monitor.PolicyVerdictNotify) pb.TrafficDirection {
 	if dn != nil && dn.Source != 0 {
 		// If the local endpoint at which the drop occurred is the same as the
-		// source of the dropped packet, we assume it was an egress flow. This
+		// source of the dropped packet, we assume it was an egress pb. This
```
Comment (gentoo-root) [inline pkg/hubble/parser/threefour/parser.go:452]: This change looks like a side effect of "replace all", it should stay as "egress flow".

**Comment by aanm** [comment]
@Archisman-Mridha I've put the PR into draft, please rebase against the latest main and changed it back from 'draft'. Thank you

**Pull request closed without merging**